core/include/gnuradio-4.0/Block.hpp

```diff
@@ -282,6 +296,11 @@ concept PublishableSpan = std::ranges::contiguous_range<T> and std::ranges::outp
 
 static_assert(PublishableSpan<traits::block::detail::dummy_output_span<float>>);
 
+struct BuiltinMessagePortPair {
+    MsgPortInNamed<"__Builtin">  input;
```

Comment:
Why the nested struct? Why not simply:

```cpp
MsgPortIn<>  msgIn;
MsgPortOut<>  msgOut;
```

Reply:
- Easier to handle one pointer in BlockModel than two
- Can not accidentally be exposed grouped with other ports thus changing indices of other ports

Reply:
At the same time these message ports require a special handling compared to the other user-defined one. I'd recommend to use/implement them equal to the user-defined ones. This would remove a special case. 

Also, there is no functional need to mandatorially group two in/out message ports. The user may choose to connect only the input or only the output.

Reply:
They are special. As we discussed before, user-defined message ports have user-defined handlers. The built-in message ports are the ports that we handle by default - that can be used for settings management etc.

The user-defined ports are not meant to be defined in pairs, but with the normal port syntax that we already have.

```
   PortMsgIn<> customMessageIn;
   PortMsgIn<> anotherMessageIn;
```

Reply:
What about having a default handler for the msgIn that invokes the optional user handler?